igvm/host.py

```diff
@@ -32,35 +36,13 @@ def get_server(hostname, servertype):
     else:
         conditions.append(Startswith(hostname + '.'))
 
-    # We want to return the server, only if it has the correct servertype,
-    # but we are not using it as a condition on the Query to be able to give
-    # better errors.
-    servers = list(Query({'hostname': Or(*conditions)}))
-
-    if not servers:
-        raise ConfigError(
-            'Server with hostname "{}" is not found.'.format(hostname)
-        )
-
-    server = servers[0]
-    for other_server in servers[1:]:
-        if other_server['servertype'] != servertype:
-            continue
-        if server['servertype'] != servertype:
-            server = other_server
-            continue
-
-        raise ConfigError(
-            'Hostname "{}" matches with multiple servers "{}" and "{}".'
-            .format(hostname, server['hostname'], other_server['hostname'])
-        )
-
-    if server['servertype'] != servertype:
-        raise ConfigError(
-            'Server "{0}" is not a "{1}".'.format(hostname, servertype)
-        )
-
-    return server
+    return Query(
+        {
+            'servertype': servertype,
+            'hostname': Or(*conditions),
+        },
+        HYPERVISOR_ATTRIBUTES if servertype == 'hypervisor' else VM_ATTRIBUTES
+    ).get()
```

Comment:
Why even have this function at all?

Reply:
I don't like this either.  The caller should always make the query before initializing the objects.  We can refactor this later.